### PR TITLE
Added more functionality 

### PR DESCRIPTION
**Pull request by AlexMollard**
Added support for PO and PP connections, Added support for lights and cameras, Slightly changed the demo project to make the window title be how long the FBX took to load for benchmarking.

### CODE REVIEW

**Comment by AlexMollard** [comment]
#11 #26

File: src/ofbx.cpp [hunk 1]
```diff
@@ -979,19 +1030,20 @@ static OptionalError<Element*> tokenize(const u8* data, size_t size, u32& versio
 	root->sibling = nullptr;
 
 	Element** element = &root->child;
-	for (;;)
+	while (true)
```
Comment (nem0) [inline src/ofbx.cpp:1033]: This causes warning on some msvc configurations, please use for (;;)

File: src/ofbx.cpp [hunk 2]
```diff
@@ -83,7 +89,19 @@ struct Video
...
+	{
+		s_message = msg;
+	}
+
+	// Format a message with printf-style arguments.
+	template <typename... Args>
+	Error(const char* fmt, Args... args)
+	{
+		char buf[1024];
+		sprintf_s(buf, fmt, args...);
```
Comment (nem0) [inline src/ofbx.cpp:102]: sprintf_s is not standard c++

File: src/ofbx.cpp [hunk 3]
```diff
@@ -1845,6 +2134,131 @@ struct OptionalError<Object*> parseTexture(const Scene& scene, const Element& el
 	return texture;
 }
 
+struct OptionalError<Object*> parseLight(Scene& scene, const Element& element, Allocator& allocator)
+{
+	LightImpl* light = allocator.allocate<LightImpl>(scene, element);
+
+	light->lightType = static_cast<Light::LightType>(resolveEnumProperty(*light, "LightType", (int)Light::LightType::POINT)); // ApertureMode
```
Comment (nem0) [inline src/ofbx.cpp:2141]: wrong comment

File: src/ofbx.cpp [hunk 4]
```diff
@@ -1535,6 +1587,198 @@ struct TextureImpl : Texture
...
+	// Shadow properties
+	const Texture* shadowTexture = nullptr;
+	bool castShadows = true;
+	Color shadowColor = {0, 0, 0};
+
+	// Area light properties
+	AreaLightShape areaLightShape = AreaLightShape::RECTANGLE;
+
+	// Barn door properties
+	float leftBarnDoor = 20.0;
```
Comment (nem0) [inline src/ofbx.cpp:1696]: unless I miss something, this is not really set anywhere

File: src/ofbx.cpp [hunk 5]
```diff
@@ -1535,6 +1587,198 @@ struct TextureImpl : Texture
...
+	double farAttenuationStart = 0.0;
+	double farAttenuationEnd = 0.0;
+
+	// Shadow properties
+	const Texture* shadowTexture = nullptr;
+	bool castShadows = true;
+	Color shadowColor = {0, 0, 0};
+
+	// Area light properties
+	AreaLightShape areaLightShape = AreaLightShape::RECTANGLE;
```
Comment (nem0) [inline src/ofbx.cpp:1693]: There's a bunch of of members that are not really set/read from fbx, like this one. Please remove

**Review by nem0 — CHANGES_REQUESTED** [review]
There are some minor issues (see review comments) and one blocker. Before I merge this please either remove unused properties (e.g. areaLightShape) or do load them from fbx.

**Comment by AlexMollard** [comment]
I have pushed a fix for all the above issues.

**Review by nem0 — COMMENTED** [review]
Seems fine, I will try to use it for a few days and merge it if I don't find any issues

**Comment by nem0** [comment]
Thanks, merged

**Pull request merged**